### PR TITLE
Initial implementation of GH Action for installing EDM

### PR DESCRIPTION
**Pull request by kitchoi**
This PR adds the GitHub action for installing EDM.

As described in https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github

These are the core files for the action functionality:
- `action.yml`: Describe and configure the action. This will be used by GitHub Actions Marketplace when we publish.
- `src/index.js`: Code required to call the script specific to the current platform, and export PATH so that edm is available on the command line 

These files are generated (see DEVELOPING.md):
- `dist/*`
- `package-lock.json`

These files are adapted from existing code:
- `src/install-edm-linux.sh`
- `src/install-edm-osx.sh`
- `src/install-edm-windows.cmd`

They are adapted from the scripts we have in ETS projects, e.g. TraitsUI.  The main changes were to make EDM version and the download directory configurable as arguments rather than as environment variables. In retrospect, I could retain using environment variables for passing parameters, but I kept it as it is now.

### CODE REVIEW

File: src/install-edm-linux.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (rahulporuri) [inline src/install-edm-linux.sh:1]: nit : can we add a copyright header here? The same applies for the other `install-edm-*` shell and cmd scripts.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+// (C) Copyright 2018-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline src/index.js]:
```suggestion
// (C) Copyright 2020 Enthought, Inc., Austin, TX
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        "sh",
+        [path.join(__dirname, "install-edm-osx.sh"), edmVersion, downloadDir]
+    )
+    // no need to update path for OSX
+
+}
+
+
+async function installEdmWindows() {
+    // Install EDM for Windowsand export the PATH such that edm can
```
Comment (rahulporuri) [inline src/index.js]:
```suggestion
    // Install EDM for Windows and export the PATH such that edm can
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "setup-edm-action",
+  "version": "1.0.0",
+  "description": "GitHub Action for installing EDM",
+  "main": "dist/index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (rahulporuri) [inline package.json]: Q : Is this necessary or can it be removed?

File: dist/index.js [hunk 5]
```diff
@@ -0,0 +1,1955 @@
+module.exports =
```
Comment (rahulporuri) [inline dist/index.js:1]: IIUC this is autogenerated. Is that correct?

File: action.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# (C) Copyright 2018-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline action.yml]:
```suggestion
# (C) Copyright 2020 Enthought, Inc., Austin, TX

```

File: LICENSE [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+The software contained in the traits/protocols/ directory is
+the pyprotocols project (http://peak.telecommunity.com/PyProtocols.html),
+it is originaly licensed under the terms of the Python Software
+Foundation License, which is compatible with the above terms.
```
Comment (rahulporuri) [inline LICENSE]:
```suggestion
```

File: LICENSE [hunk 8]
```diff
@@ -0,0 +1,33 @@
+This software is OSI Certified Open Source Software.
+OSI Certified is a certification mark of the Open Source Initiative.
+
+Copyright (c) 2006-2018, Enthought, Inc.
```
Comment (rahulporuri) [inline LICENSE]:
```suggestion
Copyright (c) 2020, Enthought, Inc.
```

**Review by rahulporuri — APPROVED** [review]
Mostly LGTM with a few comments/questions.

**Comment by kitchoi** [comment]
Sorry for the late commit: I moved the build command into the `package.json` as I noticed this is a common practice in other GitHub Actions in the marketplace.

I still find the developer workflow a bit annoying: Each time when one makes a change to the source, they need to rebuild the code in `dist` so that they are exercised by CI. It is so easy to forget that step. But if we remove that step for the development process, then we need to somehow make CI build the distribution instead, and remember to do that build step for the release as well. There is more room for automating the build step, e.g. a separate bot can do that when a commit is pushed to a PR.

File: package.json (hunk 4 above)
Comment (kitchoi) [reply package.json]: It is now removed and I added a `build` command instead.

File: dist/index.js (hunk 5 above)
Comment (kitchoi) [reply dist/index.js:1]: Yep.

**Review by rahulporuri — APPROVED** [review]
LGTM

**Pull request merged**